registry/api/v2/errors.go

```diff
@@ -151,4 +151,14 @@ var (
 		longer proceed.`,
 		HTTPStatusCode: http.StatusNotFound,
 	})
+
+	// ErrorCodeMaintenanceMode is returned when an upload can't be
+	// accepted because the registry is in maintenance mode.
+	ErrorCodeMaintenanceMode = errcode.Register(errGroup, errcode.ErrorDescriptor{
```

Comment:
I've just created an `UNAVAILABLE` error code in #831 in the `errcode` package. That might be more appropriate.